test/python/circuit/test_controlled_gate.py

```diff
@@ -768,90 +783,82 @@ def test_mcx_gates(self, num_ctrl_qubits):
         reference = np.zeros(2 ** (num_ctrl_qubits + 1))
         reference[-1] = 1
 
-        for gate in [
-            MCXGrayCode(num_ctrl_qubits),
-            MCXRecursive(num_ctrl_qubits),
-            MCXVChain(num_ctrl_qubits, False),
-            MCXVChain(num_ctrl_qubits, True),
+        for synth_circuit in [
+            synth_mcx_n_dirty_i15(num_ctrl_qubits),
+            synth_mcx_n_clean_m15(num_ctrl_qubits),
+            synth_mcx_1_clean_b95(num_ctrl_qubits),
+            synth_mcx_gray_code(num_ctrl_qubits),
+            synth_mcx_noaux_v24(num_ctrl_qubits),
         ]:
-            with self.subTest(gate=gate):
-                circuit = QuantumCircuit(gate.num_qubits)
+            with self.subTest(synth_circuit_name=synth_circuit.name):
+                circuit = QuantumCircuit(synth_circuit.num_qubits)
                 if num_ctrl_qubits > 0:
                     circuit.x(list(range(num_ctrl_qubits)))
-                circuit.append(gate, list(range(gate.num_qubits)), [])
+                circuit.compose(synth_circuit, inplace=True)
                 statevector = Statevector(circuit).data
 
                 # account for ancillas
-                if hasattr(gate, "num_ancilla_qubits") and gate.num_ancilla_qubits > 0:
+                num_ancillas = synth_circuit.num_qubits - num_ctrl_qubits - 1
+                if num_ancillas > 0:
                     corrected = np.zeros(2 ** (num_ctrl_qubits + 1), dtype=complex)
                     for i, statevector_amplitude in enumerate(statevector):
-                        i = int(bin(i)[2:].zfill(circuit.num_qubits)[gate.num_ancilla_qubits :], 2)
+                        i = int(bin(i)[2:].zfill(circuit.num_qubits)[num_ancillas:], 2)
                         corrected[i] += statevector_amplitude
                     statevector = corrected
                 np.testing.assert_array_almost_equal(statevector.real, reference)
 
     @data(5, 10, 15)
     def test_mcxvchain_dirty_ancilla_cx_count(self, num_ctrl_qubits):
```

Comment:
More tests that can be removed (I think) especially given #13961. Or actually shouldn't we keep the original test with the MCXVChain gate, that was deprecated but not yet removed? (The same comment applies to other changes in this file).

Reply:
sounds good to me... I always felt these tests were doing too much, so I'm happy that they get in the queue to be removed. Reverting to previous state in a120f94dcc5669d4ba79e9f487443692eace0d72